parser_library/src/parsing/grammar/ca_expr_rules.g4

```diff
@@ -109,9 +124,24 @@ term returns [ca_expr_ptr ca_expr]
 	}
 	| signed_num
 	{
-		collector.add_hl_symbol(token_info(provider.get_range( $signed_num.ctx),hl_scopes::number));
 		auto r = provider.get_range($signed_num.ctx);
-		$ca_expr = std::make_unique<ca_constant>($signed_num.value, r);
+		auto m_sign = $signed_num.ctx->signed_num_ch()->MINUS();
+
+		if (m_sign == nullptr)
+		{
+			collector.add_hl_symbol(token_info(r, hl_scopes::number));
+			$ca_expr = std::make_unique<ca_constant>($signed_num.value, r);
+		}
+		else
+		{
+			auto num_r = r;
+			num_r.start.column++;
+
+			collector.add_hl_symbol(token_info(num_r, hl_scopes::number));
+			auto num_expr = std::make_unique<ca_constant>(-$signed_num.value, num_r);
```

Comment:
this may in principle overflow....
what's the reason for this extra code?

Reply:
The reason is that I wanted the following expressions to throw a diagnostic:
```
&B1 SETB (+1)
&B2 SETB (-1)
```

And while the former throws diags just fine the latter doesn't without this change. The reason why `+1` throws diag is that the `+` sign is considered an operator meanwhile the `-` sign wasn't (because of `signed_num` rule). 


I wanted to take a second look at this anyway to see if there is a way to do this differently.

Reply:
I see... however I am wondering if the problem is elsewhere, because the following is valid:
```
&B1 SETB (+1 EQ +1) 
&B2 SETB (-1 EQ +1) 
&B1 SETB ((+1) EQ 0)
&B2 SETB ((-1) EQ 0)
```